mainwindow.cpp

```diff
@@ -142,9 +195,107 @@ void MainWindow::toggleOutputSplitter()
 
 bool MainWindow::eventFilter(QObject *target, QEvent *event)
 {
-    if (event->type() == QEvent::Wheel)
-        return true;
+    if (target == ui->mainOutput->viewport())
+    {
+        if (event->type() == QEvent::Resize)
+        {
+            // re position search widget when main output inner size changes
+            // this takes into account existence of vertical scrollbar
+            QResizeEvent *resizeEvent = static_cast<QResizeEvent*>(event);
+            search_widget->move(resizeEvent->size().width() - search_widget->width(), 0);
+        }
+        else if (event->type() == QEvent::Wheel)
+        {
+            // allow mouse wheel usage only in 'browsing mode'
+            if (!ui->bottomOutput->isVisible())
+                return true;
+        }
+    }
+    else
+    {
+        if (event->type() == QEvent::KeyPress)
+        {
+            QKeyEvent *keyEvent = static_cast<QKeyEvent*>(event);
+            if (ui->searchButton->isChecked())
+            {
+                // hide search widget on Escape key press
+                if (keyEvent->key() == Qt::Key_Escape)
+                    emit ui->searchButton->toggle();
+            }
+            else
+            {
+                // show search widget on Ctrl-F
+                if (keyEvent->key() == Qt::Key_F && keyEvent->modifiers() == Qt::ControlModifier)
+                    emit ui->searchButton->toggle();
+            }
+            if (target == search_input)
+            {
+                if (keyEvent->key() == Qt::Key_Return)
+                {
+                    if (keyEvent->modifiers() == Qt::ShiftModifier)
+                        emit search_prev_button->click();
+                    else
+                        emit search_next_button->click();
+                }
+            }
+
+        }
+    }
 
     // base class behaviour
     return QMainWindow::eventFilter(target, event);
 }
+
+void MainWindow::showSearchWidget(bool show)
+{
+    // record which widget had focus before showing search widget, in order
+    // to return focus to it when search widget is hidden
+    static QWidget *prevFocus = 0;
+
+    QPropertyAnimation *animation = new QPropertyAnimation(search_widget, "geometry");
```

Comment:
Come già fatto nell'event filter, anche in questo caso non c'è bisogno di lavorare sul geometry, ma puoi impostare la proprietà "pos". Sbaglio?